src/ros_filter.cpp

```diff
@@ -542,10 +542,14 @@ namespace RobotLocalization
           ROS_WARN_STREAM_DELAYED_THROTTLE(historyLength_, "Received old measurement for topic " <<
             firstMeasurementTopicName << ", but history interval is insufficiently sized to revert state and "
             "measurement queue.");
-          restoredMeasurementCount = 0;
+          measurementQueue_.pop();
```

Comment:
I think this would be easier to read this way:

``` c++
if (revertTo(firstMeasurementTime - 1e-9))
{
  restoredMeasurementCount = static_cast<int>(measurementQueue_.size()) - originalCount;
  break;
}

RF_DEBUG("ERROR: history interval is too small to revert to time " << firstMeasurementTime << "\n");
ROS_WARN_STREAM_DELAYED_THROTTLE(historyLength_, "Received old measurement for topic " <<
  firstMeasurementTopicName << ", but history interval is insufficiently sized to revert state and "
  "measurement queue.");
measurementQueue_.pop();
```

Reply:
I can do that if you think it's better, I was just leaning toward reducing the delta.

Reply:
Yes, that makes sense. I guess you'll send this PR upstream after some testing on the robots.

I leave it up to you.